lib/services/third_party_services/audio_player_helper.dart

```diff
@@ -0,0 +1,105 @@
+import 'dart:async';
+
+import 'package:just_audio/just_audio.dart';
+
+/// Abstract Class for providing audio player
+abstract class AudioPlayerHelper {
+  ///This method works as an intializer for Audio Player,
+  ///It will allow us to pass the url for the Audio
+  ///Returns true if Url is loaded otherwise false
+  Future<bool> setFile({required String file});
+
+  ///This method works as an intializer for Audio Player,
+  Future<void> setAudioSource({required AudioSource file});
+
+  /// This method will be a listener to the playerStateStream
+  /// which will listen the audio stream being player
+  StreamSubscription<PlayerState> playerStateStream();
+
+  /// This method will be a listener to the positionStream which
+  /// will listen to the current DurationPosition of the audio track
+  StreamSubscription<Duration> positionStream();
+
+  /// This method will be a listener to the bufferedPositionStream which
+  /// will listen to the duration position where the audio is being buffered
+  StreamSubscription<Duration> bufferedPositionStream();
+
+  /// This method will be a listener to the durationStream which
+  /// will listen to the DurationStream being player
+  StreamSubscription<Duration?> durationStream();
+
+  /// This method will be responsible for destroying the audio player instances from the memory
+  void destroyAudioPlayer();
+
+  /// This method will be responsible for Pausing the Audio
+  Future<void> pauseAudio();
+
+  /// This method will be responsible for Playing the Audio
+  Future<void> playAudio();
+
+  /// This method will be responsible for seeking the audio when dragged forward of reverse
```

Comment:
No Input documentation here.

Reply:
No parameter is going in this function. That is why it has no input documentation.